lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -477,7 +477,7 @@ sub mark_job_linked {
     return undef if !$job || ($referer->path_query =~ /^\/?$/);
     my $comments = $job->comments;
     return undef if $comments->find({text => {like => 'label:linked%'}});
-    my $user = $self->result_source->schema->resultset('Users')->search({username => 'system'})->first;
+    my $user = $self->result_source->schema->resultset('Users')->find({username => 'system'});
```

Comment:
There's no unique constraint (only on a combination of `username` and `provider`) so this should be limited to one row.

Reply:
What do you mean with "this should be limited to one row"?

Reply:
You cannot do queries returning more than one row with `find` (see also @Amrysliu's comment). One of the lines you've changed already limits the result to one row: `$last_job_id = $schema->resultset('Jobs')->find(undef, {rows => 1, order_by => {-desc => 'id'}})->id;`

Reply:
> You cannot do queries returning more than one row with find

Well, actually one can. https://metacpan.org/pod/DBIx::Class::ResultSet#find says "If no such constraint is found, find currently defaults to a simple search->(\%column_values) which may or may not do what you expect. Note that this fallback behavior may be deprecated in further versions. If you need to search with arbitrary conditions - use "search". If the query resulting from this fallback produces more than one row, a warning to the effect is issued, though only the first row is constructed and returned as $result_object."

so in short: The first match will be returned and a warning is printed.

However, I thought we would only apply search+first when we already know there is only one match. I consider it risky to just take the "first" match when we don't even expected others to be there. I wonder if it wouldn't be better to make "find" fail hard if there are more than one matches unless a force parameter is given or something

Reply:
> so in short: The first match will be returned and a warning is printed.

I know, and a warning is not acceptable so I said you cannot use `find` with more than one row.

Likely search+fist was used in context where it didn't matter which row was returned exactly or where it was an accepted limitation that you can only get the first.

In this particular case we want the system user so we should include its provider into the query. I suppose the provider of the system user is just an empty string.

Reply:
I changed the query to look for `provider => ''` additionally now